src/CodeGen_ARM.cpp

```diff
@@ -952,6 +953,31 @@ void CodeGen_ARM::visit(const Load *op) {
             Value *elt = builder->CreateExtractValue(call, {(unsigned int)offset});
             results.push_back(elt);
         }
+#else
+
+        llvm::Type *load_return_type = llvm_type_of(op->type.with_lanes(intrin_lanes*stride->value));
```

Comment:
Ideally yes, but then CodeGen_LLVM will need to generate the same pattern to get LLVM to generate the proper intrinsics.  From what I can tell it doesn't do that for now.
As a first step, I would add this code generation pass in CodeGen_ARM, let it stick (make sure there are no failures/regressions, and removing the <37 pass is safe), and then integrate it into CodeGen_LLVM.